pkg/topology/kademlia/kademlia.go

```diff
@@ -1247,7 +1288,18 @@ func (k *Kad) Disconnected(peer p2p.Peer) {
 
 	k.connectedPeers.Remove(peer.Address)
 
-	k.waitNext.SetTryAfter(peer.Address, time.Now().Add(k.opt.TimeToRetry))
+	k.connRetryBackMtx.Lock()
+	if k.connRetryBackoff[peer.Address.String()] == 0 {
+		k.connRetryBackoff[peer.Address.String()] = 1
+	} else {
+		k.connRetryBackoff[peer.Address.String()] *= 2
+	}
+
+	multiplier := k.connRetryBackoff[peer.Address.String()]
+
+	k.connRetryBackMtx.Unlock()
+
```

Comment:
Could we have this logic in individual functions because code seems to be repeating?